core/store/src/trie/mem/updating.rs

```diff
@@ -610,175 +569,286 @@ impl<'a, M: ArenaMemory> MemTrieUpdate<'a, M> {
         key: &[u8],
         value: FlatStateValue,
     ) -> Result<(), StorageError> {
-        self.insert_impl(key, GenericTrieValue::MemtrieOnly(value))
+        self.generic_insert(0, key, GenericTrieValue::MemtrieOnly(value))
+    }
+}
+
+impl<
+        'a,
+        N: std::fmt::Debug,
+        V: std::fmt::Debug + HasValueLength,
+        T: GenericTrieUpdate<'a, N, V>,
+    > GenericTrieUpdateInsertDelete<'a, N, V> for T
```

Comment:
nit: Could we change this to `impl GenericTrieUpdateInsertDelete<...> where N: std::fmt::Debug, ....` format for readability?